src/adapter/src/catalog/migrate.rs

```diff
@@ -182,6 +209,567 @@ pub(crate) async fn migrate(
+
+                // A reference to the source that will be included in the table statement
+                let source_ref =
+                    RawItemName::Id(item.id.to_string(), new_source_name.clone(), None);
+
+                let columns = if col_names.is_empty() {
+                    TableFromSourceColumns::NotSpecified
+                } else {
+                    TableFromSourceColumns::Named(col_names)
+                };
+
+                // All source tables must have a `details` option, which is a serialized proto
+                // describing any source-specific details for this table statement.
+                let details = match &conn {
+                    // For kafka sources this proto is currently empty.
+                    CreateSourceConnection::Kafka { .. } => SourceExportStatementDetails::Kafka {},
+                    CreateSourceConnection::LoadGenerator { .. } => {
+                        // Since these load generators are single-output we use the default output.
+                        SourceExportStatementDetails::LoadGenerator {
+                            output: LoadGeneratorOutput::Default,
+                        }
+                    }
+                    _ => unreachable!("match determined above"),
+                };
+                let mut table_with_options = vec![TableFromSourceOption {
+                    name: TableFromSourceOptionName::Details,
+                    value: Some(WithOptionValue::Value(Value::String(hex::encode(
+                        details.into_proto().encode_to_vec(),
+                    )))),
+                }];
+
+                // Move over the IgnoreKeys option if it exists.
+                if let Some(i) = with_options
+                    .iter()
+                    .position(|opt| opt.name == CreateSourceOptionName::IgnoreKeys)
+                {
+                    let option = with_options.remove(i);
+                    table_with_options.push(TableFromSourceOption {
+                        name: TableFromSourceOptionName::IgnoreKeys,
+                        value: option.value,
+                    });
+                };
+                // Move over the Timeline option if it exists.
+                if let Some(i) = with_options
+                    .iter()
+                    .position(|opt| opt.name == CreateSourceOptionName::Timeline)
+                {
+                    let option = with_options.remove(i);
+                    table_with_options.push(TableFromSourceOption {
+                        name: TableFromSourceOptionName::Timeline,
+                        value: option.value,
+                    });
+                }
+
+                // Generate the same external-reference that would have been generated
+                // during purification for single-output sources.
+                let external_reference = match &conn {
+                    // For kafka sources this proto is currently empty.
```

Comment:
Is this a copy and paste? I don't really understand what is supposed to be empty.

Reply:
ah yep copy pasta